master/buildbot/www/auth.py

```diff
@@ -113,17 +113,19 @@ def __init__(self, header=None, headerRegex=None, **kwargs):
         if header is not None:
             self.header = header
         if headerRegex is not None:
-            self.headerRegex = re.compile(unicode2bytes(headerRegex))
+            self.headerRegex = re.compile(headerRegex)
 
     @defer.inlineCallbacks
     def maybeAutoLogin(self, request):
         header = request.getHeader(self.header)
         if header is None:
-            raise Error(403, b"missing http header " + self.header + b". Check your reverse proxy config!")
+            raise Error(403, b"missing http header " + self.header +
+                             b". Check your reverse proxy config!")
         res = self.headerRegex.match(header)
         if res is None:
-            raise Error(
-                403, b'http header does not match regex! "' + header + b'" not matching ' + self.headerRegex.pattern)
+            err = 'http header does not match regex! "' + header + \
+                  '" not matching ' + self.headerRegex.pattern
+            raise Error(403, unicode2bytes(err))
```

Comment:
This change is still wrong and unnecessary.  You should just revert this entire line.  `header` and `headerRegex` need to be bytes